File: Clg DSA/quicksort.c
#include <stdio.h>
void quick(int arr[],int low ,int high)
{ int i,j,pivot,temp;
    if(low<high)
    { i=low;
        j=high;
        pivot=low;
        while(i<j)
        {
            while (arr[i]<=arr[pivot] && i<high)
            {
                 i++;
                 while (arr[j]>arr[pivot])
                 {
                    j--;
                    if(i<j)
                    {
                        temp=arr[i];
                        arr[i]=arr[j];
                        arr[j]=temp;
                    }
                 }
                 
            }
            
        }
        temp=arr[pivot];
        arr[pivot]=arr[j];
        arr[j]=temp;
        quick(arr,low,j-1);
        quick(arr,j+1,high);

    }


}
int main()
{ int arr[]={7,6,5,4,1};
int n=5;
 quick(arr,0,n-1);
 for(int i=0;i<n;i++)
 {
    printf("%d",arr[i]);
 }


}

File: Sorting/quicksort.java
public class quicksort 
{ public static int  partition(int arr[],int low,int high)
    {
        int pivot=arr[high];
        int i=low-1;
        for(int j=low;j<high;j++)
        {
            if(arr[j]<pivot)
            {
                i++;
            
            //swwaping
            int temp=arr[i];
            arr[i]=arr[j];
            arr[j]=temp;        
            }}i++;
            int temp=arr[i];
            arr[i]=pivot;
            arr[high]=temp;
            return i;
    }
    

    public static void quicksort(int arr[],int low,int high)
    {
        if(low<high)
        {
            int pidx=partition(arr, low, high);
            quicksort(arr, low, pidx-1);
            quicksort(arr, pidx+1, high);
        }
    }
public static void main(String[] args) {
    int arr[]={3,2,1,5};
    int n=arr.length;
quicksort(arr, 0, n-1);
//print
for(int i=0;i<=n;i++)
{
    System.out.println(arr[i]+"");
}
System.out.println();
}
}
